streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -2138,18 +2100,16 @@ object ZStream extends ZStreamPlatformSpecific {
     managed(fa.toManaged_)
 
   /**
-   * Creates a stream from an [[InputStream]].
+   * Creates a stream from a [[Pull]].
    */
-  final def fromInputStream[R, E, A](is: InputStream[R, E, A]): ZStream[R, E, A] =
-    fromInputStreamManaged(ZManaged.succeed(is))
+  final def fromPull[R, E, A](pull: Pull[R, E, A]): ZStream[R, E, A] =
+    fromPullManaged(ZManaged.succeed(pull))
 
   /**
-   * Creates a stream from a scoped [[InputStream]].
+   * Creates a stream from a scoped [[Pull]].
    */
-  final def fromInputStreamManaged[R, E, A](is: ZManaged[R, E, InputStream[R, E, A]]): ZStream[R, E, A] =
-    new ZStream[R, E, A] {
-      def process = is
-    }
+  final def fromPullManaged[R, E, A](pull: ZManaged[R, E, Pull[R, E, A]]): ZStream[R, E, A] =
```

Comment:
Let's rename this to `def apply`

Reply:
`def apply` and keep curly braces or go with the new `ZStream()`?

Reply:
The definition here has to be `def apply(pull: ..) = new ZStream(pull)`, but in the call sites you can then do:

```scala
ZStream {
  <expression that returns pull>
}
```